### PR TITLE
AX-504 [Pool Page - My Liquidity] Implement My Liquidity BLoc

### PR DESCRIPTION
**Pull request by bekchanovj**
# Description
Set up MyLiquidity Bloc and added unit tests for the usecase and the bloc. 

Fixes Jira Ticket # 
504
https://athletex.atlassian.net/browse/AX-504

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Has been tested using unit tests. 

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have assigned 2 reviewers to check my work
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import 'package:ax_dapp/service/BlockchainModels/LiquidityPosition.dart';
+import 'package:decimal/decimal.dart';
+import 'package:fpdart/fpdart.dart';
+
+class GetAllLiquidityInfoUseCase {
+  final SubGraphRepo _graphRepo;
+
+  GetAllLiquidityInfoUseCase(this._graphRepo);
+
+  Future<Either<Success, SubgraphError>> fetchAllLiquidityTokensInfoForWallet(
```
Comment (ketonkss4) [inline lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart:13]: this sounds kind of awkward maybe 
`fetchInfoForAllLiquidityToken` or `fetchAllLiquidityTokenDetails`

File: lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    final Decimal reserve0 = Decimal.parse(liquidityPosition.pair.reserve0);
+    final Decimal reserve1 = Decimal.parse(liquidityPosition.pair.reserve1);
+    final Decimal lpTokenPairBalance =
+        Decimal.parse(liquidityPosition.liquidityTokenBalance);
+    final Decimal lpTokenTotalSupply =
+        Decimal.parse(liquidityPosition.pair.totalSupply!);
+    final Decimal shareOfPool = (lpTokenPairBalance / lpTokenTotalSupply)
+        .toDecimal(
+            scaleOnInfinitePrecision:
+                37); //TODO: How to round irrational numbers?
```
Comment (ketonkss4) [inline lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart:76]: tag a ticket in the backlog for this or it will never get done

File: lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    final String token0LpAmount = (shareOfPool * reserve0).toString();
+    final String token1LpAmount = (shareOfPool * reserve1).toString();
+    final String apy = '0'; // TODO: Implement APY calculation
```
Comment (ketonkss4) [inline lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart:79]: tag a ticket in the backlog for this

File: lib/pages/pool/MyLiqudity/bloc/MyLiquidityBloc.dart [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      LoadEvent event, Emitter<MyLiquidityState> emit) async {
+    emit(state.copyWith(status: BlocStatus.loading));
+    final String? walletAddress = controller.getWalletAddress();
+    try {
+      if (walletAddress != null) {
+        final response =
+            await repo.fetchAllLiquidityTokensInfoForWallet(walletAddress: walletAddress);
+        final isSuccess = response.isLeft();
+        if (isSuccess) {
+          final myLiquidityItemInfos =
```
Comment (ketonkss4) [inline lib/pages/pool/MyLiqudity/bloc/MyLiquidityBloc.dart:30]: `liquidityDetailItemList` or `liquiditityInfoItemList`... information does not have a dimension of countability so it sounds awkward to say infos... 

also aren't these `pools`? Instead of items cant we say `LiquidityPoolDetailsList`?

File: test/MyLiquidityBloc_test.dart [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+import 'MyLiquidityBloc_test.mocks.dart';
+
+@GenerateMocks([GetAllLiquidityInfoUseCase, GetWalletAddressUseCase])
+void main() {
+  var mockRepoUseCase = MockGetAllLiquidityInfoUseCase();
+  var mockWalletAddressUseCase = MockGetWalletAddressUseCase();
+  var myLiquidityBloc = MyLiquidityBloc(
+      repo: mockRepoUseCase, controller: mockWalletAddressUseCase);
+  final testWalletAddress = "testWalletAddress";
+  final List<MyLiquidityItemInfo> testMyLiquidityItemInfo = [
```
Comment (ketonkss4) [inline test/MyLiquidityBloc_test.dart:19]: this is overkill, your logic doesn't even use this data, mock an object or objects and put them in a list.

**Review by ketonkss4 — CHANGES_REQUESTED** [review]

File: test/GetAllLiquidityInfoUseCase_test.dart [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+            apy: "0"));
+    assert(myLiquidityItemInfos![1] ==
+        MyLiquidityItemInfo(
+            token0Name: "AthleteX",
+            token1Name: "Juan Soto Linear Short Token",
+            token0Symbol: "AX",
+            token1Symbol: "JSST1010",
+            token0Address: "0x5617604ba0a30e0ff1d2163ab94e50d8b6d0b0df",
+            token1Address: "0xce44443d4f652fc6c48f62258c75278e11909d6a",
+            lpTokenPairAddress: "0x49c62531a60feb3ccc8f85a6bb897bce53036e5b",
```
[the remark this answers is not in the document]
Comment (bekchanovj) [reply test/GetAllLiquidityInfoUseCase_test.dart:57]: Yeah, I was doing only one and it passed, but when I added the rest it threw an error. But I'll clean it up and keep edge cases there. At least the ones that I know of, like one number with 38 digits after floating-point.

File: lib/pages/pool/MyLiqudity/bloc/MyLiquidityBloc.dart (hunk 4 above)
Comment (bekchanovj) [reply lib/pages/pool/MyLiqudity/bloc/MyLiquidityBloc.dart:30]: Yeah, `LiquidityPoolDetailsList` sounds good.

File: lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart (hunk 1 above)
Comment (bekchanovj) [reply lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart:13]: Changed it to fetchAllLiquidityPositions.

File: lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart (hunk 2 above)
Comment (bekchanovj) [reply lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart:76]: done

File: lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart (hunk 3 above)
Comment (bekchanovj) [reply lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart:79]: done

File: test/GetAllLiquidityInfoUseCase_test.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (ketonkss4) [reply test/GetAllLiquidityInfoUseCase_test.dart]: Yeah that's a good idea

**Review by ketonkss4 — APPROVED** [review]

**Review by Friendlygeckos — APPROVED** [review]
